src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,215 @@
+import io.sphere.sdk.http.AsyncHttpClientAdapter;
+import io.sphere.sdk.http.HttpClient;
+import io.sphere.sdk.retry.RetryAction;
+import io.sphere.sdk.retry.RetryContext;
+import io.sphere.sdk.retry.RetryPredicate;
+import io.sphere.sdk.retry.RetryRule;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClient;
+import org.asynchttpclient.DefaultAsyncHttpClientConfig;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+/**
+ * To create a Sphere Client with retry logic which computes a exponential backoff time delay in milliseconds.
+ * And handle all the configurations for the creation of client.
+ */
+public final class RetryableSphereClientWithExponentialBackoff {
+    protected static final long DEFAULT_TIMEOUT = 30000;
+    protected static final long INITIAL_RETRY_DELAY = 200;
+    protected static final TimeUnit DEFAULT_TIMEOUT_TIME_UNIT = TimeUnit.MILLISECONDS;
+    protected static final int MAX_RETRIES = 5;
+    protected static final int MAX_PARALLEL_REQUESTS = 20;
+    private static final int[] DEFAULT_RETRY_ERROR_STATUS_CODES = {500, 502, 503, 504};
+
+    private SphereClientConfig clientConfig;
+    private long timeout;
+    private long initialDelay;
+    private TimeUnit timeUnit;
+    private int maxRetries;
+    private int maxParallelRequests;
+    private int[] retryErrorStatusCodes;
+
+    private RetryableSphereClientWithExponentialBackoff(@Nonnull final SphereClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        this.timeout = DEFAULT_TIMEOUT;
+        this.timeUnit = DEFAULT_TIMEOUT_TIME_UNIT;
+        this.initialDelay = INITIAL_RETRY_DELAY;
+        this.maxRetries = MAX_RETRIES;
+        this.maxParallelRequests = MAX_PARALLEL_REQUESTS;
+        this.retryErrorStatusCodes = DEFAULT_RETRY_ERROR_STATUS_CODES;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param clientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the timeout value.
+     * @param timeout - build with timeout value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeout value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeout(final long timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value.
```

Comment:
Java docs are not explaining much, what is the value and why is needed? for example, it's a value in milliseconds. 

Similar to others, but you could skip this comment.

Reply:
Updated at few places, Let me know if anything can be updated to make more readable javadocs. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/b68b095b731c1cbd8b717184decd4d8ed9693762